Corona-Warn-App/src/test/java/de/rki/coronawarnapp/datadonation/survey/server/DataDonationServerTest.kt

```diff
@@ -0,0 +1,89 @@
+package de.rki.coronawarnapp.datadonation.survey.server
+
+import android.content.Context
+import de.rki.coronawarnapp.datadonation.OneTimePassword
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.EdusOtp
+import io.kotest.matchers.shouldBe
+import io.mockk.MockKAnnotations
+import io.mockk.clearAllMocks
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.runBlocking
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import testhelpers.BaseIOTest
+import testhelpers.BaseTest
+import java.io.File
+import java.util.UUID
+
+class DataDonationServerTest : BaseTest() {
+    @MockK lateinit var dataDonationApi: DataDonationApiV1
+    @MockK lateinit var context: Context
+
+    private lateinit var webServer: MockWebServer
+    private lateinit var serverAddress: String
+
+    private val testDir = File(BaseIOTest.IO_TEST_BASEDIR, this::class.java.simpleName)
+    private val cacheDir = File(testDir, "cache")
+
+    @BeforeEach
+    fun setup() {
+        MockKAnnotations.init(this)
+        testDir.mkdirs()
+        testDir.exists() shouldBe true
+        every { context.cacheDir } returns cacheDir
+
+        webServer = MockWebServer()
```

Comment:
The webserver isn't needed here as this class works by mocking `DataDonationApiV1`.

Reply:
thanks.